x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -807,38 +808,66 @@ public void updateConnectorLastSyncStats(UpdateConnectorLastSyncStatsAction.Requ
     }
 
     /**
-     * Updates the is_native property of a {@link Connector}. It always sets the {@link ConnectorStatus} to
-     * CONFIGURED.
+     * Updates the is_native property of a {@link Connector}. It sets the {@link ConnectorStatus} to
+     * CONFIGURED when connector is in CONNECTED state to indicate that connector needs to reconnect.
      *
      * @param request  The request for updating the connector's is_native property.
      * @param listener The listener for handling responses, including successful updates or errors.
      */
     public void updateConnectorNative(UpdateConnectorNativeAction.Request request, ActionListener<UpdateResponse> listener) {
         try {
             String connectorId = request.getConnectorId();
+            boolean isNative = request.isNative();
 
-            final UpdateRequest updateRequest = new UpdateRequest(CONNECTOR_INDEX_NAME, connectorId).doc(
-                new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.INDEX)
-                    .id(connectorId)
-                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-                    .source(
-                        Map.of(
-                            Connector.IS_NATIVE_FIELD.getPreferredName(),
-                            request.isNative(),
-                            Connector.STATUS_FIELD.getPreferredName(),
-                            ConnectorStatus.CONFIGURED.toString()
-                        )
-                    )
+            getConnector(connectorId, listener.delegateFailure((l, connector) -> {
 
-            );
-            client.update(updateRequest, new DelegatingIndexNotFoundActionListener<>(connectorId, listener, (l, updateResponse) -> {
-                if (updateResponse.getResult() == UpdateResponse.Result.NOT_FOUND) {
-                    l.onFailure(new ResourceNotFoundException(connectorNotFoundErrorMsg(connectorId)));
+                String indexName = getConnectorIndexNameFromSearchResult(connector);
+
+                // Ensure attached content index is prefixed correctly
+                if (isNative && indexName != null && isValidManagedConnectorIndexName(indexName) == false) {
```

Comment:
```suggestion
                // Ensure attached content index is prefixed correctly
                boolean doesNotHaveContentPrefix = indexName != null && isValidManagedConnectorIndexName(indexName) == false
                
                if (isNative && doesNotHaveContentPrefix) {
```

Rationale: [Cognitive load is what matters: complex conditionals](https://github.com/zakirullin/cognitive-load?tab=readme-ov-file#complex-conditionals)

Reply:
Good read! Thank you for linking